include/http/request_handler.js

```diff
@@ -538,57 +538,74 @@ module.exports = function RequestHandlerModule(pb) {
         return false;
     };
 
-    /**
-     * Serves up a 404 page when the path specified by the incoming request does 
-     * not exist. This function <b>WILL</b> close the connection.
-     * @method serve404
-     */
-    RequestHandler.prototype.serve404 = function() {
-
-        var NotFound  = require(path.join(pb.config.docRoot, 'plugins/pencilblue/controllers/error/404.js'))(pb);
-        var cInstance = new NotFound();
-        this.doRender({}, cInstance, {});
+  /**
+   * Serves up a 404 page when the path specified by the incoming request does
+   * not exist. This function <b>WILL</b> close the connection.
+   * @method serve404
+   */
+  RequestHandler.prototype.serve404 = function() {
+
+    var self = this;
+    var NotFound;
+    var settings = pb.SettingServiceFactory.getService(pb.config.settings.use_memory, pb.config.settings.use_cache, this.site);
+    settings.get('active_theme', function(err, theme) {
+
+      try{
+        NotFound = require(path.join(pb.config.docRoot, 'plugins/' + theme + '/controllers/error/404.js'))(pb);
+      } catch(e){
+        NotFound = require(path.join(pb.config.docRoot, 'plugins/pencilblue/controllers/error/404.js'))(pb);
+      }
+
+      var cInstance = new NotFound();
+      self.doRender({}, cInstance, {});
+
+      if (pb.log.isSilly()) {
+        pb.log.silly("RequestHandler: No Route Found, Sending 404 for URL=" + this.url.href);
+      }
```

Comment:
Neat trick to save you a couple of lines and keep performance.  The logging functions provide the ability to pass in parameters to the log statement such they can be formatted in.  Adding parameters to the function stack doesn't cost much.  About as much as doing the if statement. 
```
pb.log.silly('RequestHandler: No Route Found, Sending 404 for URL=%s', this.url.href);
```
The above will only do the replacement if the log level is "silly".